File: JS/day2.js
const input = require("../uploads/day2");
const fs = require('fs')

let data = input.split("\n").map((each) => each.split(" "));

// part one
let answerA = {x: 0, y:0, final:0}
data.forEach((each) => {
  switch (each[0]) {
    case "forward":
      answerA.x += +each[1];
      break;
    case "down":
      answerA.y += +each[1];
      break;
    case "up":
      answerA.y -= +each[1];
      break;
    default:
      break;
  }
});

answerA.final = answerA.x * answerA.y
console.log(answerA.final);

// part two
let answerB = {x: 0, y:0, aim:0, final:0}

data.forEach((each) => {
  switch (each[0]) {
    case "forward":
      answerB.x += +each[1];
      answerB.y += answerB.aim * +each[1]
      break;
    case "down":
      answerB.aim += +each[1];
      break;
    case "up":
      answerB.aim -= +each[1];
      break;
    default:
      break;
  }
});

answerB.final = answerB.x * answerB.y
console.log(answerB.final);


fs.writeFileSync('../out/day2.json', JSON.stringify({answerA, answerB}) , {encoding:'utf-8', flag:'w'})

File: Python/day2.py
with open('uploads/day2.js', 'r') as file:
    filestring = file.read()
filestring = filestring.split('`')[1].splitlines()

# Part one
answerA = {'x':0, 'y':0, 'final':0} 
inputArray = []

for i in range(len(filestring)):
  [direction, num] = filestring[i].split(' ')
  if direction == 'forward':
    answerA['x'] += int(num)
  elif direction == 'up':
    answerA['y'] -= int(num)
  elif direction == 'down':
    answerA['y'] += int(num)

answerA['final'] = answerA['x'] * answerA['y']
print(answerA['final'])

# Part two
answerB = {'x':0, 'aim':0, 'y':0, 'final':0} 
inputArray = []

for i in range(len(filestring)):
  [direction, num] = filestring[i].split(' ')
  if direction == 'forward':
    answerB['x'] += int(num)
    answerB['y'] += int(num) * answerB['aim']
  elif direction == 'up':
    answerB['aim'] -= int(num)
  elif direction == 'down':
    answerB['aim'] += int(num)

answerB['final'] = answerB['x'] * answerB['y']
print(answerB['final'])